raimitigations/dataprocessing/imputer/imputer.py

```diff
@@ -179,25 +218,43 @@ def _revert_encoding(
         missing_value_cols_no_impute: list,
         non_valid_cols: list,
     ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        Attempts to revert the encoding of all categorical data in the set,
+        whether it was imputed or not.
+
+        :param transf_df: pandas dataframe post possible encoding and imputation;
+
+        :param df: original pandas dataframe before imputation or encoding;
+
+        :param df_to_transf: pandas dataframe post possible encoding, pre imputation;
+
+        :param cat_cols_missing_value_impute: list of categorical columns that were imputed;
+
+        :param cat_cols_no_missing_value: list of categorical columns that had no missing
+            values but needed to be encoded to be part of the imputation process;
+
+        :param missing_value_cols_no_impute: list of columns that contain missing values
+            but weren't specified in col_impute;
+
+        :param non_valid_cols: list of columns excluded at fit time, possibly because
+            they are categorical and enable_encoder=False;
+
+        :return: resulting pandas dataframe or np.ndarray
+        :rtype: pd.dataframe or np.ndarray
+        """
+
         # attempt inverse_transform encoded imputed columns.
-        try:
-            if self.enable_encoder is True:
-                decoded_transf_df = self.ordinal_encoder.inverse_transform(transf_df)
-                transf_df[cat_cols_missing_value_impute] = decoded_transf_df[cat_cols_missing_value_impute]
-                self.print_message(f"\nImputed categorical columns' encoding was reverse transformed.")
-        except IndexError:
-            self.print_message(
-                f"\nImputed categorical columns' encoding was not reverse transformed."
-                + f"Note that encoded columns are not guaranteed to reverse transform if they have imputed values.\n"
-            )
-            pass
+        if self.enable_encoder is True:
+            decoded_transf_df = _inverse_transform_ordinal_encoder_with_new_values(self.inverse_mapping, transf_df)
+            transf_df[cat_cols_missing_value_impute] = decoded_transf_df[cat_cols_missing_value_impute]
+            self.print_message(f"\nImputed categorical columns' reverse encoding transformation complete.")
 
         # reverse cols with missing values not in col_impute to their original values.
         transf_df[missing_value_cols_no_impute] = df[missing_value_cols_no_impute]
 
         # inverse_transform cat cols not in col_impute.
         if self.enable_encoder is True and len(cat_cols_no_missing_value) > 0:
-            decoded_df = self.ordinal_encoder.inverse_transform(df_to_transf)
+            decoded_df = _inverse_transform_ordinal_encoder_with_new_values(self.inverse_mapping, df_to_transf)
             transf_df[cat_cols_no_missing_value] = decoded_df[cat_cols_no_missing_value]
```

Comment:
The previous comment applies here as well. Actually, it applies to any case where you assign multiple columns from one df to another.

Reply:
Originally, I had split them before because inverse transforming columns with imputed values could have failed and needed exception handling, but categorical data that wasn't imputed but encoded should always go back to the original state. Now that I changed the inverse transform function, I should be able to combine them.